core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -342,19 +342,24 @@ class GroupMetadataManager(val brokerId: Int,
             (topicPartition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET, "", Errors.NONE))
           }.toMap
         } else {
-            if (topicPartitionsOpt.isEmpty) {
-              // Return offsets for all partitions owned by this consumer group. (this only applies to consumers that commit offsets to Kafka.)
-              group.allOffsets.map { case (topicPartition, offsetAndMetadata) =>
-                (topicPartition, new OffsetFetchResponse.PartitionData(offsetAndMetadata.offset, offsetAndMetadata.metadata, Errors.NONE))
-              }
-            } else {
-              topicPartitionsOpt.getOrElse(Seq.empty[TopicPartition]).map { topicPartition =>
-                group.offset(topicPartition) match {
-                  case None => (topicPartition, new OffsetFetchResponse.PartitionData(OffsetFetchResponse.INVALID_OFFSET, "", Errors.NONE))
-                  case Some(offsetAndMetadata) =>
-                    (topicPartition, new OffsetFetchResponse.PartitionData(offsetAndMetadata.offset, offsetAndMetadata.metadata, Errors.NONE))
+            topicPartitionsOpt match {
+              case None =>
+                // Return offsets for all partitions owned by this consumer group. (this only applies to consumers
+                // that commit offsets to Kafka.)
+                group.allOffsets.mapValues { offsetAndMetadata =>
```

Comment:
`mapValues` returns a view over the original collection (annoyingly), which means that the returned map can change after it's returned if `group.allOffsets` is updated. This doesn't seem to be what we want here.